sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -456,8 +456,17 @@ object JdbcUtils extends Logging {
 
         case StringType =>
           (array: Object) =>
-            array.asInstanceOf[Array[java.lang.String]]
-              .map(UTF8String.fromString)
+            array match {
+              case _: Array[java.lang.String] =>
+                array.asInstanceOf[Array[java.lang.String]]
+                  .map(UTF8String.fromString)
+              case _: Array[java.util.UUID] =>
+                array.asInstanceOf[Array[java.util.UUID]]
```

Comment:
Can this also be covered by the following `Array[java.lang.Object]`?

Reply:
Yes, Object can cover the UUID since it's a Superclass in Java. Should we not clearly distinguish between classes? Would you mind giving some comments? Thanks!

Reply:
Having a separate case for UUID is more clear to me and maybe easy to debug in the future. However I'm not sure how long this matching cases can be grown.

Reply:
Ok. Then we remove this UUID pattern and only have used `Array[java.lang.Object]`.